shm/Ship.cpp

```diff
@@ -34,6 +34,6 @@ void Ship::unload(Cargo* cargo) {
 
 void Ship::nextDay() {
     if (delegate_) {
-        delegate_->payCrew(crew_ * salaryPerWorker); 
+        delegate_(crew_ * salaryPerWorker);
```

Comment:
This `delegate_(crew_ * salaryPerWorker);` is not as clear as previous version `delegate_->payCrew(crew_ * salaryPerWorker);` 

In old one it's clear that there is some object which calls the payCrew method. In new approach, for me it looks like some kind of constructor or something like that. Because we have a member delegate_ which turns out to be a function, not an attribute. But naming convention with underline at the end suggest that it is class field. Maybe the name should be changed to different one?

Reply:
Changed